java/README.md

```diff
@@ -0,0 +1,220 @@
+  is acquired by using the OpenTelemetry API specifying the name and version of the library instrumenting the
+  instrumented library or application to be monitored.
+  ```java
+    private static Tracer getTracer() {
+        tracer = openTelemetry.getTracer(INSTRUMENTATION_LIBRARY_NAME, INSTRUMENTATION_VERSION);         
+        return tracer;
+    }
+  ```
+  Note: the ```name``` and ```optional version``` of the tracer are purely informational. All Tracers that are created
+  by a single OpenTelemetry instance will interoperate, regardless of name.
+
+* #### Create a nested span, add attributes
+  To create a basic span, you only need to specify the name of the span. The start and end time of the span is
+  automatically set by the OpenTelemetry SDK. Most of the time, we want to correlate spans for nested operations. In
+  OpenTelemetry spans can be created freely and it’s up to the implementor to annotate them with attributes specific to
+  the represented operation. Attributes provide additional context on a span about the specific operation it tracks,
+  such as results or operation properties. For the ```main``` method to call the ```child``` method, the spans could be
+  manually linked in the following way:
+    ```java
+      public static void main(String[] args) throws InterruptedException {
+
+        /*this will make sure that a proper service.name attribute is set on all the
+         spans/metrics.*/
+        System.setProperty(OTEL_RESOURCE_ATTRIBUTES_KEY, OTEL_RESOURCE_ATTRIBUTES_VALUE);
+
+        /*tracer must be acquired, which is responsible for creating spans and interacting with the Context*/
+        tracer = getTracer();
+
+        /*an automated way to propagate the parent span on the current thread*/
+        for (int index = 0; index < 3; index++) {
+            /*create a span by specifying the name of the span. The start and end time of the span is automatically set by the OpenTelemetry SDK*/
+            Span parentSpan = tracer.spanBuilder("parentSpan").startSpan();
+            logger.info("In parent method. TraceID : {}", parentSpan.getSpanContext().getTraceIdAsHexString());
+
+            /*put the span into the current Context*/
+            try (Scope scope = parentSpan.makeCurrent()) {
+
+                /*annotate the span with attributes specific to the represented operation, to provide additional context*/
+                parentSpan.setAttribute("parentIndex", index);
+                childMethod(parentSpan);
+            } catch (Throwable throwable) {
+                parentSpan.setStatus(StatusCode.ERROR, "Something wrong with the parent span");
+            } finally {
+                /*closing the scope does not end the span, this has to be done manually*/
+                parentSpan.end();
+            }
+        }
+
+        /*sleep for a bit to let everything settle*/
+        Thread.sleep(2000);
```

Comment:
Just curious if the API will let you block until everything is settled.  Thread.sleep() is tricky to use because how do you know that you have slept long enough?

Reply:
That is just an estimate. You are right, we can not say for sure if we have slept enough.